Emby.Server.Implementations/Updates/InstallationManager.cs

```diff
@@ -131,85 +173,62 @@ public async Task<IList<PackageInfo>> GetPackages(string manifestName, string ma
-            var destVersion = dest[0].VersionNumber;
-
-            while (d < dLength)
-            {
-                if (sourceVersion.CompareTo(destVersion) >= 0)
-                {
-                    if (s < sLength)
-                    {
-                        sourceVersion = source[++s].VersionNumber;
-                    }
-                    else
-                    {
-                        // Append all of destination to the end of source.
-                        while (d < dLength)
-                        {
-                            source.Add(dest[d++]);
-                        }
-
-                        break;
-                    }
-                }
-                else
-                {
-                    source.Insert(s++, dest[d++]);
-                    if (d >= dLength)
-                    {
-                        break;
-                    }
-
-                    sLength++;
-                    destVersion = dest[d].VersionNumber;
-                }
-            }
-        }
-
         /// <inheritdoc />
         public async Task<IReadOnlyList<PackageInfo>> GetAvailablePackages(CancellationToken cancellationToken = default)
         {
             var result = new List<PackageInfo>();
             foreach (RepositoryInfo repository in _config.Configuration.PluginRepositories)
             {
-                if (repository.Enabled)
+                if (repository.Enabled && repository.Url != null)
                 {
-                    // Where repositories have the same content, the details of the first is taken.
-                    foreach (var package in await GetPackages(repository.Name, repository.Url, cancellationToken).ConfigureAwait(true))
+                    // Where repositories have the same content, the details from the first is taken.
+                    foreach (var package in await GetPackages(repository.Name ?? "Unnamed Repo", repository.Url, true, cancellationToken).ConfigureAwait(true))
                     {
-                        if (!Guid.TryParse(package.guid, out var packageGuid))
+                        if (!Guid.TryParse(package.Guid, out var packageGuid))
                         {
                             // Package doesn't have a valid GUID, skip.
                             continue;
                         }
 
-                        for (var i = package.versions.Count - 1; i >= 0; i--)
+                        var existing = FilterPackages(result, package.Name, packageGuid).FirstOrDefault();
+
+                        // Remove invalid versions from the valid package.
+                        for (var i = package.Versions.Count - 1; i >= 0; i--)
                         {
+                            var version = package.Versions[i];
+
+                            var plugin = _pluginManager.GetPlugin(packageGuid, version.VersionNumber);
+                            // Update the manifests, if anything changes.
```

Comment:
Why update the manifest on disk? I think it shouldn't be changed from what is provided from the plugin repo

Reply:
_version_ contains the package's version information as defined in the repo. (line 184 - GetPackages)

This checks the repo's version against what we have locally, and updated out local manifest if the MaxAbi or TargetAbi has changed.

This effectively allows us to update these values in the repository if we find that a version no longer works with a release - thus stopping it from being loaded next time. (Similar to what we had with 10.7)

Was going to recommend that for the release notes for 10.8, the user runs the goes to repository screen just prior to upgrading. That way the local manifests will be set to the latest correct values - and we shouldn't have the same issues as 10.6.x -> 10.7. Hope that makes sense